### ISSUE TITLE
[outerloop] [NativeAOT] Segmentation fault in System.Runtime.InteropServices.Tests

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/runtime#95414]
### Error Blob

```json
{
  "ErrorMessage": "Segmentation fault      (core dumped) ./System.Runtime.InteropServices.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop",
  "BuildRetry": false,
  "ErrorPattern": "",
  "ExcludeConsoleLog": true
}
```

### Reproduction Steps

@dotnet/ilc-contrib @dotnet/interop-contrib PTAL

- Affected PR: https://github.com/dotnet/runtime/pull/94868 via `/azp run runtime-nativeaot-outerloop`
- Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=483727
- Job: https://dev.azure.com/dnceng-public/public/_build/results?buildId=483727&view=logs&j=20e6f673-0b0c-5c85-377c-5d941d4c8ee3&t=5801e1cc-59c5-5e47-8484-142b970afd58&l=55
- Queue: `linux_musl-x64 Release NativeAOT_Libs`
- Log file: https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-94868-merge-67c5e1f44298409881/System.Runtime.InteropServices.Tests/1/console.cb42cc43.log?helixlogtype=result
- Output: 
```
===========================================================================================================
/root/helix/work/workitem/e /root/helix/work/workitem/e
DOTNET_DbgEnableMiniDump is set and the createdump binary does not exist: ./createdump
Running assembly:System.Runtime.InteropServices.Tests, Version=9.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51
[SKIP] System.Runtime.InteropServices.Tests.GetHINSTANCETests.GetHINSTANCE_ModuleBuilder_ReturnsSameInstance
[SKIP] System.Runtime.InteropServices.Tests.Int16Tests.WriteInt16_NotReadable_ThrowsArgumentException
[SKIP] System.Runtime.InteropServices.Tests.Int16Tests.ReadInt16_NotReadable_ThrowsArgumentException
[SKIP] System.Runtime.InteropServices.Tests.CreateWrapperOfTypeTests.CreateWrapperOfType_NullType_ThrowsArgumentNullException
[SKIP] System.Runtime.InteropServices.Tests.GetComObjectDataTests.GetComObjectData_NullObj_ThrowsArgumentNullException
[SKIP] System.Runtime.InteropServices.Tests.GetComObjectDataTests.GetComObjectData_NullKey_ThrowsArgumentNullException
[SKIP] System.Runtime.InteropServices.Tests.GetComObjectDataTests.GetComObjectData_NonComObjectObj_ThrowsArgumentNullException
./RunTests.sh: line 179:    22 Segmentation fault      (core dumped) ./System.Runtime.InteropServices.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop -notrait category=failing -xml testResults.xml $RSP_FILE
/root/helix/work/workitem/e
----- end Wed Nov 29 16:21:17 UTC 2023 ----- exit code 139 ----------------------------------------------------------
exit code 139 means SIGSEGV Illegal memory access. Deref invalid pointer, overrunning buffer, stack overflow etc. Core dumped.
```

 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=483727
**Error message validated:** `Segmentation fault      (core dumped) ./System.Runtime.InteropServices.Tests -notrait category=IgnoreForCI -notrait category=OuterLoop`
**Result validation: :x:** Known issue did not match with the provided build.
**Validation performed at:** 11/29/2023 6:05:02 PM UTC

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

**Comment by jtschuster** [comment]
Looks like another related to https://github.com/dotnet/runtime/pull/94485. I'll revert that and investigate the issues.

**Comment by carlossanlop** [comment]
Thank you @jtschuster!

This was discovered via the unrelated PR https://github.com/dotnet/runtime/pull/94868 where we are enabling XUnitLogChecker for most libraries tests. I decided to temporarily skip enabling the tool for NativeAOT tests as they required some different handling, which is why the only message we see is "segmentation fault" and nothing else, unfortunateoly. But we intend to get the XUnitLogChecker tool enabled in NativeAOT as well after I merge that PR, and you should be able to see the output of the core dumps directly in the log files.

**Comment by jkotas** [comment]
Fixed by #95415
